### PR TITLE
Add `watch` option

### LINKED ISSUE #75: Trigger `onDidChange` when the file is altered
[Issuehunt badges](https://issuehunt.io/r/sindresorhus/conf/issues/75)

See https://github.com/sindresorhus/conf/pull/47 for an initial attempt that went stale. Please take into account all the feedback given there if you intend to take on the issue.

Part of this task is also to bring the docs over to `electron-store` to resovle https://github.com/sindresorhus/electron-store/issues/39.

---

> Note: This is not an easy issue. It requires you to have good JavaScript and Node.js experience. I expect you to do a good job. Instead of asking to many questions, present solutions. The point of a issue bounty is to give me less work, not more. 🙌

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Sponsors (Total: $80.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($80.00)

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/conf/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

### PR DESCRIPTION
**Pull request by yaodingyd**
Fix #75

### CODE REVIEW

**Comment by yaodingyd** [comment]
working on the tests

File: index.d.ts [hunk 1]
```diff
@@ -163,6 +163,13 @@ declare namespace Conf {
 
 		*/
 		readonly accessPropertiesByDotNotation?: boolean;
+
+		/**
+		Watch for any changes in the config file and call the callback for `onDidChange` if set. This is usedful if there are multiple processes changing the same config file.
```
Comment (saylestyler) [inline index.d.ts]: usedful -> useful

**Comment by yaodingyd** [comment]
The only failing test is on Mac node 8 where I find `fs.watch` is not working at all. Running this
```
const fs = require('fs');
const path = require('path');
const tempy = require('tempy');
const Conf = require('.');

const cwd = tempy.directory();
const conf1 = new Conf({cwd, watch: true});
const conf2 = new Conf({cwd});
conf1.set('foo', '👾');
console.log(conf1.get('foo'));

const checkFoo = () => {
	console.log('change detected');
};

conf1.onDidChange('foo', checkFoo);
conf2.set('foo', '🐴');
console.log(conf1.get('foo'));

fs.writeFileSync(path.join(cwd, 'config.json'), JSON.stringify({foo: '🦄'}));
console.log(conf1.get('foo'));
```

doesn't show the "change detected" console log on my Mac.

**Comment by yaodingyd** [comment]
@zeke Can you help me test Mac node 8 by running the above test file under Conf directory?

**Comment by sindresorhus** [comment]
@yaodingyd I tried your above script on Node.js 8 and 12 on macOS and I only get: 

```
👾
🐴
🦄
```

**Comment by yaodingyd** [comment]
@sindresorhus Thanks for helping test! So up to this point all the tests give pretty consistent results: `fs.watch` works on all system/node.js combination except macOS with Node.js 8. I'm afraid we have to use chokidar to make it work cross-platform.

**Comment by sindresorhus** [comment]
@yaodingyd Just document that the `watch` option doesn't work on Node.js 8 on macOS.

**Comment by yaodingyd** [comment]
Quirks documented.

**Comment by sindresorhus** [comment]
@yaodingyd Can you fix the merge conflicts?

**Comment by sindresorhus** [comment]
Looks good :)

**Comment by sindresorhus** [comment]
Can you open a PR on electron-store to document it there too?

**Pull request merged**